WordPress/Classes/ViewRelated/Post/AztecPostViewController.swift

```diff
@@ -274,8 +274,10 @@ class AztecPostViewController: UIViewController {
             originalPostStatus = postStatus
         }
 
-        // TODO: Determine if user can actually publish to site or not
-        let context = PostEditorStateContext(originalPostStatus: originalPostStatus, userCanPublish: true, delegate: self)
+        // Self-hosted non-Jetpack blogs have no capabilities, so we'll default
+        // to showing Publish Now instead of Submit for Review.
+        let userCanPublish = self.post.blog.capabilities != nil ? self.post.blog.isPublishingPostsAllowed() : true
```

Comment:
It looks like isPublishingPostsAllowed will handle nil capabilities properly so all you need is:

```swift
let userCanPublish = self.post.blog.isPublishingPostsAllowed()
```